### ISSUE TITLE
[SR-10684] LLDB fails to get context or resolve identifier in protocol extensions and generic functions

### ISSUE THREAD

**Issue by jpsim** [swiftlang/llvm-project#4339]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-10684      |
   |Radar             | None         |
   |Original Reporter | @jpsim      |
   |Type              | Bug    |
   |Status            | Resolved        |
   |Resolution        | Done    |
        

Attachment: [Download](https://user-images.githubusercontent.com/2727770/164963288-3dca2f00-2c4d-49b1-bcdc-0753b034fcb2.gz)

      
<details>
  <summary>Environment</summary>

macOS Mojave 10.14.5 (18F132)  
Xcode 10.2.1 (10E1001)  
Apple Swift version 5.0.1 (swiftlang-1001.0.82.4 clang-1001.0.46.5)  
Target: x86_64-apple-darwin18.6.0

</details>

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 0         |
   |Component/s       | LLDB for Swift    |
   |Labels            | Bug        |
   |Assignee          | None      |
   |Priority          | Medium      |

   

   md5: 84982b8a80323f0a719a6dcf7d77af5b

  </details>

**Issue Description:**

It's very common to get errors such as

``` java
po thing
error: <EXPR>:3:1: error: use of unresolved identifier 'thing'
thing
^~~~~
```

or

``` java
po thing
error: <EXPR>:1:11: error: use of undeclared type '$__lldb_context'
extension $__lldb_context {
          ^~~~~~~~~~~~~~~

error: <EXPR>:19:5: error: use of unresolved identifier '$__lldb_injected_self'
    $__lldb_injected_self.$__lldb_wrapped_expr_75(
    ^~~~~~~~~~~~~~~~~~~~~
```

when using LLDB in scopes such as protocol extensions and generic functions.

I've attached a very minimal sample projects for both iOS and a command line Swift app that demonstrate these issues reliably. They both fail in slightly different ways.

There's also a screen recording demonstrating the steps I'm taking to trigger these issues.

**Comment by slavapestov** [comment]
These were all fixed in the 5.1 branch.

**Comment by jpsim** [comment]
Thanks for the quick reply and thanks to you and other contributors for fixing this!
